Symfony/CS/Fixer/Contrib/MethodArgumentDefaultValueFixer.php

```diff
@@ -0,0 +1,167 @@
+                break;
+            }
+
+            if (
+                $this->isDefaultArgumentAfterIndex($tokens, $nextRelevantIndex - 1) &&
+                $nextRelevantIndex - 1 < $lastNonDefaultArgumentIndex
+            ) {
+                $this->removeDefaultArgument($tokens, $nextRelevantIndex);
+            }
+            $examinedIndex = $nextRelevantIndex;
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return int|null
+     */
+    private function getLastNonDefaultArgumentIndex(Tokens $tokens, $index)
+    {
+        $nextRelevantTokenIndex = $tokens->getNextTokenOfKind($index, $this->variableOrTerminatorTokens);
+
+        if (null === $nextRelevantTokenIndex) {
+            return;
+        }
+
+        $lastNonDefaultArgumentIndex = null;
+
+        while ($tokens[$nextRelevantTokenIndex]->isGivenKind(T_VARIABLE)) {
+            if ($tokens[$tokens->getNextMeaningfulToken($nextRelevantTokenIndex)]->equalsAny($this->argumentTerminatorTokens)) {
+                $lastNonDefaultArgumentIndex = $nextRelevantTokenIndex;
+            }
+
+            $nextRelevantTokenIndex = $tokens->getNextTokenOfKind($nextRelevantTokenIndex, $this->variableOrTerminatorTokens);
+        }
+
+        return $lastNonDefaultArgumentIndex;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function hasDefaultValueAfterIndex(Tokens $tokens, $index)
+    {
+        $nextTokenIndex = $tokens->getNextTokenOfKind($index, $this->defaultValueTokens);
+        $nextToken = $tokens[$nextTokenIndex];
+
+        return $nextToken->equals('=');
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isDefaultArgumentAfterIndex(Tokens $tokens, $index)
+    {
+        $nextTokenIndex = $tokens->getNextTokenOfKind($index, $this->immediateDefaultValueTokens);
+        $nextToken = $tokens[$nextTokenIndex];
+
+        return $nextToken->equals('=');
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $nextVariableIndex
+     */
+    private function removeDefaultArgument(Tokens $tokens, $nextVariableIndex)
+    {
+        $currentIndex = $nextVariableIndex;
+
+        $prevMeaningfulTokenIndex = $tokens->getPrevMeaningfulToken($nextVariableIndex);
+        $nextMeaningfulTokenIndex = $tokens->getNextTokenOfKind($nextVariableIndex, array(array(T_STRING, 'null'), ',', ')'));
+
+        if (
+            $tokens[$nextMeaningfulTokenIndex]->equals(array(T_STRING, 'null')) &&
+            !$tokens[$prevMeaningfulTokenIndex]->equalsAny($this->argumentBoundaryTokens)
+        ) {
+            return; //ignore typehinted null default values
+        }
+
+        while (!$tokens[$currentIndex + 1]->equalsAny($this->argumentTerminatorTokens)) {
+            $tokens[++$currentIndex]->clear();
```

Comment:
Please add this test case:

```php
function foo($foo = array(array(1)), $bar) {}
```

Should fail because `$this->argumentTerminatorTokens` contains `)`.